mne/source_estimate.py

```diff
@@ -383,6 +383,12 @@ def _make_stc(data, vertices, src_type=None, tmin=None, tstep=None,
         src_type = _get_src_type(src=None, vertices=vertices,
                                  warn_text=warn_text)
 
+    if vector and isinstance(data, tuple):
+        kernel = True
+        data, sens_data = data[0], data[1]
+    else:
+        kernel = False
```

Comment:
This looks really weird to me. So depending on the fact that vector is a tuple or not, we create this `kernel` flag and we modify our data container. Then, we overwirte (again) the data container just before the calls.

If nothing else, I would write:
```diff
             for i, d, n in zip(range(data.shape[0]), data, source_nn):
                 data_rot[i] = np.dot(n.T, d)
-            data = data_rot
+
             if kernel:
-                data = (data, sens_data)
-            stc = VectorSourceEstimate(data, vertices=vertices, tmin=tmin,
-                                       tstep=tstep, subject=subject)
+                stc = VectorSourceEstimate(
+                    data=(data_rot, sens_data), vertices=vertices, tmin=tmin,
+                    tstep=tstep, subject=subject
+                )
+            else:
+                stc = VectorSourceEstimate(
+                    data=data_rot, vertices=vertices, tmin=tmin,
+                    tstep=tstep, subject=subject
+                )
+
```

or something like this to make clear that I'm calling the same constructor with the very same parameters:
```py
            kwargs = dict(vertices=vertices, tmin=tmin, tstep=tstep, subject=subject)

            if kernel:
                stc = VectorSourceEstimate(data=(data_rot, sens_data), *kwargs)
            else:
                stc = VectorSourceEstimate(data=data_rot, *kwargs)
```

Reply:
or if  you really wanted to use `data` and given that you already rewrite it with `data_rot`:

```diff
             for i, d, n in zip(range(data.shape[0]), data, source_nn):
                 data_rot[i] = np.dot(n.T, d)
-            data = data_rot
-            if kernel:
-                data = (data, sens_data)
+
+            data = (data_rot, sens_data) if kernel else data_rot
             stc = VectorSourceEstimate(data, vertices=vertices, tmin=tmin,
                                        tstep=tstep, subject=subject)
```